src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java

```diff
@@ -153,69 +138,80 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
     return future;
   }
 
+  public CompletableFuture<JsonObject> handleGetRequest(String endpoint) {
+      CompletableFuture<JsonObject> future = new VertxCompletableFuture<>(ctx);
+      try {
+        logger.debug("Calling GET {}", endpoint);
+        httpClient.request(HttpMethod.GET, endpoint, okapiHeaders)
+          .thenApply(response -> {
+            logger.debug("Validating response for GET {}", endpoint);
+            return HelperUtils.verifyAndExtractBody(response);
+          })
+          .thenAccept(body -> {
+            if (logger.isDebugEnabled()) {
+              logger.debug("The response body for GET {}: {}", endpoint, nonNull(body) ? body.encodePrettily() : null);
+            }
+            future.complete(body);
+          })
+          .exceptionally(t -> {
+            logger.error(EXCEPTION_CALLING_ENDPOINT_MSG, t, HttpMethod.GET, endpoint);
+            future.completeExceptionally(t);
+            return null;
+          });
+      } catch (Exception e) {
+        logger.error(EXCEPTION_CALLING_ENDPOINT_MSG, e, HttpMethod.GET, endpoint);
+        future.completeExceptionally(e);
+      }
+      return future;
+    }
+
   public CompletableFuture<String> lookupLocationId(String location) {
-    try {
-      String query = HelperUtils.encodeValue(String.format(CQL_CODE_STRING_FMT, location));
-      return httpClient.request("/locations?query=" + query, okapiHeaders)
-        .thenApply(HelperUtils::verifyAndExtractBody)
+      String query = HelperUtils.encodeValue(String.format(CQL_CODE_STRING_FMT, location), logger);
+      String endpoint = String.format(LOCATIONS_ENDPOINT+QUERY, query);
+      return handleGetRequest(endpoint)
         .thenApply(HelperUtils::extractLocationId)
         .exceptionally(t -> {
           logger.error("Exception looking up location id", t);
           return null;
         });
-    } catch (Exception e) {
-      logger.error("Exception calling lookupLocationId", e);
-      throw new CompletionException(e);
-    }
+
   }
   public CompletableFuture<List<String>> lookupMaterialTypeId(String materialType) {
-    try {
-      String query = HelperUtils.encodeValue(String.format("name==\"%s\"", materialType));
-      return httpClient.request("/material-types?query=" + query, okapiHeaders)
-        .thenApply(HelperUtils::verifyAndExtractBody)
+      String query = HelperUtils.encodeValue(String.format("name==%s", materialType), logger);
```

Comment:
If we use `LoggerFactory`, `logger` as a parameter need not be required. This applies to `lookupVendorId` and `lookupPaymentStatusId` below as well.

Reply:
Same here, we need PostGobiOrdersHelper to pass its own logger